### ISSUE TITLE
`clippy::unsafe_removed_from_name` should not fire for `use SomeUnsafe as _` imports

### ISSUE THREAD

**Issue by Qix-** [rust-lang/rust-clippy#16768]
### Summary

When importing a trait anonymously (via `use SomeTrait as _`), `clippy::unsafe_removed_from_name` will fire if the trait name has `Unsafe` in it.

The _spirit_ of this lint, as I understand it, is that renaming something from `Unsafe` to _another_ name that lacks the `Unsafe` is undesirable (I agree). However, importing it just to get the side effects of the trait being in scope (anonymously, via `as _`) is _not_ renaming it, and is arguably not "dropping" the `Unsafe` from it at all.

While I like this lint, writing preludes is kind of a headache as this lint has to be disabled file-wide, which might cause false-negatives in the future. I'd suggest amending the logic such that `as _` imports of items with `Unsafe` in the name do _not_ trigger this lint.

### Lint Name

unsafe_removed_from_name

### Reproducer

```rust
pub mod base {
    pub trait SomeUnsafeFoo {
        fn foo(&self);
    }
}

struct MyFoo;

const _: () = {
    impl base::SomeUnsafeFoo for MyFoo {
        fn foo(&self) {
            println!("MyFoo!");
        }
    }
};

pub mod prelude {
    pub use super::base::SomeUnsafeFoo as _;
}

pub fn main() {
    use prelude::*;

    MyFoo.foo();
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=1a13f09813b5266c8357374283c92014

### Version

```text
rustc 1.95.0-nightly (d940e5684 2026-01-19)
binary: rustc
commit-hash: d940e56841ddcc05671ead99290e35ff2e98369f
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 21.1.8
```
